patches/net/minecraft/world/effect/MobEffectInstance.java.patch

```diff
@@ -50,17 +83,50 @@
                  .result();
      }
  
-@@ -348,6 +_,15 @@
+@@ -348,6 +_,48 @@
          this.blendState.setImmediate(this);
      }
  
-+    private final java.util.Set<net.neoforged.neoforge.common.EffectCure> cures = com.google.common.collect.Sets.newIdentityHashSet();
++    private final net.neoforged.neoforge.common.util.SetReference<net.neoforged.neoforge.common.EffectCure, java.util.Set<net.neoforged.neoforge.common.EffectCure>> cures =
++            new net.neoforged.neoforge.common.util.SetReference<>(com.google.common.collect.Sets.newIdentityHashSet());
```

Comment:
I don't think this is a great idea. Use Concurrent Collections or synchronize on the private field.